CONTRIBUTING.md

```diff
@@ -1,9 +1,190 @@
 
-Feel free to work on any section of code that you believe you can improve. More importantly, remember to thoroughly test all your classes and functions, and to provide clear, detailed comments within your code. This not only aids others in using your library, but also facilitates future maintenance and further development.
+Feel free to work on any section of code that you believe you can improve. More importantly, remember to thoroughly test all your classes and functions, and to provide clear, detailed comments within your code. This not only aids others in using the library, but also facilitates future maintenance and further development.
+
+## Contributing to the code
+
+### Contribution workflow cycle
+
+In order to contribute, you will need to do the following:
+
+1) Create your own branch
+2) Make sure that tests pass and code coverage is maintained
+3) Open a Pull Request
+
+The NeMoS package follows the [GitHub Flow](https://www.gitkraken.com/learn/git/best-practices/git-branch-strategy#github-flow-branch-strategy) workflow. In essence, this means no one is allowed to
+push to the `main` branch and all development happens in separate feature branches that are then merged into `main` once we have determined they are ready. When enough changes have accumulated, we 
+put out a new release, adding a new tag which increments the version number, and upload the new release to PyPI. 
+
+
+#### Creating a development environment
+
+You will need a local installation of `nemos` which keeps up-to-date with any changes you make. To do so, you will need to clone `nemos` and checkout a new branch:
+
+1) Clone the `nemos` repo to your local machine
+
+```bash
+git clone https://github.com/flatironinstitute/nemos.git
+cd nemos
+```
+
+2) Install `nemos` in editable mode with developer dependencies  
+```bash
+pip install -e .[dev]
+```
+
+> **_NOTE:_**  In order to install `nemos` in editable mode you will need a Python virtual environment. Please see our documentation [here](https://nemos.readthedocs.io/en/latest/installation/) that 
+> provides guidance on how to create and activate a virtual environment.
+
+
+#### Creating a new branch
+
+As mentioned previously, each feature in `nemos` is worked on in a separate branch. This allows multiple people developing multiple features simultaneously, without interfering with each other's work. To create
+your own branch, run the following from within your `nemos` directory:
+
+```bash
+# switch to the main branch
+git checkout main
+# update your main branch
+git pull origin main
+# create and switch to a new branch
+git checkout -b my_feature_branch
+```
+
+After you have made changes on this branch, add and commit them when you are ready:
+
+```bash
+# stage the changes
+git add src/nemos/the_changed_file.py
```

Comment:
what's your opinion of using:
git add -u .

for adding any updated file?

Reply:
My only thought is that sometimes there are files that we don't want people to commit. For example, I often test things in a juypter notebook but I would never want to commit that. I think it is better to have people do `git status` and then selectively choose the correct files.

Reply:
are the notebook are part of the repo?

Reply:
I agree it's a bad idea to tell people to use `git add .` (though I wasn't aware of `-u`, that's a handy flag, and avoids many of the potential problems). but I agree with Caitlin that we want people to be selective in what they add